### PR TITLE
Suppress `Unresolved reference` inspection in test mods in packages with cyclic dev-dependencies

### PR DESCRIPTION
**Pull request by vlad20012**

### CODE REVIEW

**Comment by vlad20012** [comment]
This fixes 254 false positives in tokio and  214 false positives in amethyst

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsDocAndAttributeOwner.kt [hunk 1]
```diff
@@ -458,21 +458,26 @@ fun RsAttributeOwnerStub.isEnabledByCfgSelf(crate: Crate): Boolean {
...
     val attributeStub = if (this is RsDocAndAttributeOwner) attributeStub else this as RsAttributeOwnerStub
     if (attributeStub?.mayHaveCfg == false) return ThreeValuedLogic.True
 
-    val crate = crateOrNull ?: containingCrate ?: return ThreeValuedLogic.True // TODO: maybe unknown?
-    val evaluator = CfgEvaluator.forCrate(crate)
+    val evaluator = lazyEvaluator.createEvaluator(this) ?: return ThreeValuedLogic.True // TODO: maybe unknown?
 
     val rawMetaItems = attributeStub?.rawMetaItems ?: rawMetaItems
 
+
```
Comment (dima74) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsDocAndAttributeOwner.kt:480]:
```suggestion
```

File: src/test/kotlin/org/rust/lang/core/psi/RsIsUnderCfgTestTest.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    private fun doTest() {
+        val (element, data) = findElementAndDataInEditor<RsElement>()
+
+        val expected = when (data) {
+            "true" -> true
+            "false" -> false
+            else -> error("Unknown value `$data`")
+        }
```
Comment (dima74) [inline src/test/kotlin/org/rust/lang/core/psi/RsIsUnderCfgTestTest.kt:92]:
```suggestion
        val expected = data.toBooleanStrict()
```

**Review by dima74 — APPROVED** [review]

File: src/test/kotlin/org/rust/lang/core/psi/RsIsUnderCfgTestTest.kt (hunk 2 above)
Comment (vlad20012) [reply src/test/kotlin/org/rust/lang/core/psi/RsIsUnderCfgTestTest.kt:92]: `toBooleanStrict` is considered unresolved on CI, for some reason :(

File: src/main/kotlin/org/rust/lang/core/psi/ext/CfgUtils.kt [hunk 3]
```diff
@@ -78,6 +77,45 @@ fun PsiElement.getCodeStatus(crate: Crate?): RsCodeStatus {
     return RsCodeStatus.CODE
 }
 
+/**
+ * `#[cfg(test)]` or `#[test]`
+ */
+val PsiElement.isUnderCfgTest: Boolean
+    get() = ancestorOrSelf<RsElement>()?.isUnderCfgTest == true
+
+private val RsElement.isUnderCfgTest: Boolean
```
[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/psi/ext/CfgUtils.kt:86]: Now it should work with expanded elements

**Comment by vlad20012** [comment]
bors r=dima74

**Pull request merged**